RELEASE-NOTES.md

```diff
@@ -1,5 +1,27 @@
 # Prefect Release Notes
 
+## Release 2.10.17
```

Comment:
Making we can take this opportunity to celebrate our new contributors since we have more than usual?

Otherwise we could highlight our new and improved tutorial!

Reply:
I added a highlight for the tutorial.